duet/duet_test.py

```diff
@@ -402,6 +410,43 @@ async def func():
         assert isinstance(exc_info.value.__context__, impl.Interrupt)
         assert exc_info.value.__suppress_context__
 
+    @duet.sync
+    async def test_timeout(self):
+        start = time.time()
+        with pytest.raises(TimeoutError):
+            async with duet.timeout(0.2):
+                await duet.AwaitableFuture()
+        assert abs((time.time() - start) - 0.2) < 0.02
```

Comment:
I increased the timeouts to 0.5 seconds and the tolerance when checking them to 0.2 seconds. I originally thought about injecting a clock so we could control time more carefully in tests, but I think it will mostly complicate things for not much benefit. However, I'm open to being convinced otherwise so we could certainly revisit this in the future if desired.

Reply:
One hacky way would be to set `time_func` as an attribute on `Scheduler` then add `get_current_scheduler().time_func = my_time`. Anyway, we can play around with this in the future :)